translate.py

```
@@ -0,0 +1,1685 @@
+        # Load snapshot once per source file
+        snapshot_path = get_snapshot_path(source_xml, config.repo_root)
+        snapshot = load_snapshot(snapshot_path)
+
+        # Read source entries once per source file
+        source_entries = read_source_strings(source_xml)
+
+        # Determine if snapshot needs update
+        snapshot_needs_update = _snapshot_needs_update(snapshot, source_entries)
+
+        if snapshot_needs_update and snapshot:
+            logger.debug(f"  Source strings changed since last snapshot")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)